src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String SEPARATOR = ",";
+    private static final int FIRST = 0;
+    private static final int SECOND = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] data = readFromFile(fromFileName);
+        String report = generateReport(data);
```

Comment:
It would be better to have the readFromFile method focus solely on reading raw content from the file, without performing any data transformations or calculations. Right now, the method reads, parses, and calculates some initial values, returning an array with two elements.

This conflates multiple responsibilities and makes the function less reusable and harder to test. It's recommended to let all the parsing and calculations happen in generateReport. This would also improve code readability and maintainability by clearly separating the logic.

Reply:
I am absolutely agree with it 
but especially in this case i will transfer redundant data and have to made redundant function to operate this data

Reply:
your `readFromFile` method not only reads the file, it also makes calculations that must be performed in this `generateReport` method, and this is already a violation of the development principles of single responsibility, so it is better to redo it as Oleh recommended